src/Sylius/Bundle/ShopBundle/Twig/Component/Common/FastestDeliveryTimeComponent.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ShopBundle\Twig\Component\Common;
+
+use Sylius\Bundle\UiBundle\Twig\Component\TemplatePropTrait;
+use Sylius\Bundle\ShopBundle\Provider\FastestDeliveryTimeProvider;
+use Sylius\TwigHooks\LiveComponent\HookableLiveComponentTrait;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Symfony\UX\LiveComponent\Attribute\AsLiveComponent;
+use Symfony\UX\LiveComponent\DefaultActionTrait;
+use Symfony\UX\TwigComponent\Attribute\ExposeInTemplate;
+
+#[AsLiveComponent]
+final class FastestDeliveryTimeComponent
+{
+    use DefaultActionTrait;
+    use HookableLiveComponentTrait;
+    use TemplatePropTrait;
+
+    private const TRANS_EXACT = 'sylius.ui.delivery_time.exact';
+
+    private const TRANS_RANGE = 'sylius.ui.delivery_time.range';
+
+    public function __construct(
+        private readonly FastestDeliveryTimeProvider $fastestDeliveryTimeProvider,
```

Comment:
We can inject the interface here and make this component independent of fastest delivery, which could simply be the default implementation in Sylius. And by creating a composite with priorities, we could simply create a new service with a higher priority in the end application to be used instead of the default one. As one of the arguments, we could define an `array $context` to be able to pass additional information to the provider 🤔 

```suggestion
        private readonly DeliveryTimeProviderInterface $deliveryTimeProvider,
```